packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/FormEventProcessor.java

```diff
@@ -175,15 +180,23 @@ public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDesc
     public void refresh(ChangeDescription changeDescription) {
         if (this.shouldRefresh(changeDescription)) {
             Form form = this.refreshForm();
-
             this.currentForm.set(form);
+            this.emitNewForm(changeDescription);
+        } else if (changeDescription.getKind().equals(ChangeKind.RELOAD_REPRESENTATION) && changeDescription.getSourceId().equals(this.currentForm.get().getId())) {
+            Optional<Form> reloadedForm = this.representationSearchService.findById(this.editingContext, this.currentForm.get().getId(), Form.class);
+            if (reloadedForm.isPresent()) {
+                this.currentForm.set(reloadedForm.get());
+                this.emitNewForm(changeDescription);
+            }
+        }
+    }
```

Comment:
Would that really work for forms? Wouldn't that reload the empty form from the database and send it back as is to the frontend making everything go blank?

Could you add the relevant tests for all representations to prove that your contribution is working please?